Content.Shared/Timing/UseDelaySystem.cs

```diff
@@ -0,0 +1,89 @@
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Content.Shared.Cooldown;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
+
+namespace Content.Shared.Timing;
+
+public sealed class UseDelaySystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+    private HashSet<UseDelayComponent> _activeDelays = new();
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var toRemove = new RemQueue<UseDelayComponent>();
+
+        foreach (var delay in _activeDelays)
+        {
+            MetaDataComponent? metaData = null;
+
+            if (Deleted(delay.Owner, metaData) ||
+                delay.CancellationTokenSource?.Token.IsCancellationRequested == true)
+            {
+                toRemove.Add(delay);
+                continue;
+            }
+
+            if (Paused(delay.Owner, metaData)) continue;
+
+            delay.Elapsed += frameTime;
+
```

Comment:
Just incrementally increasing this in a shared system will cause issues for clients doing prediction.

You either need to give this component state handling code & mark as dirty, so it gets reset after prediction (which is icky, because the server would try and send a new component state to clients every tick), or instead of basing it off the elapsed time, maybe just compare `IGameTiming.CurrentTime` with `startTime + delay`, so that you don't update the component every tick.

Actually now that I think about it you might just need to have the state-handling stuff in general for the cancellation logic to work properly with prediction (so it can un-cancel when resetting state). That or maybe just remove the cancellation functionality for now, given that nothing seems to use it?

Reply:
Though if nothing currently uses it in shared code, maybe just move it to server and add a TODO so people don't accidentally make that mistake when moving it to shared later.

Reply:
Probably better to move to server for now given this is already an issue with the current SpawnTimer approach and it shouldn't be regressing at the very least.